src/main/java/com/puppycrawl/tools/checkstyle/xpath/AttributeNode.java

```diff
@@ -0,0 +1,60 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.xpath;
+
+import net.sf.saxon.type.Type;
+
+/**
+ * Represents attribute of the element.
+ *
+ * @author Timur Tibeyev
+ */
+public class AttributeNode extends AbstractNode {
+    /** The name of the attribute. */
+    private final String name;
+
+    /**
+     * Creates a new {@code AttributeNode} instance.
+     *
+     * @param name name of the attribute
+     */
+    public AttributeNode(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Returns type of the node.
+     * @return node kind
+     */
+    @Override
+    public int getNodeKind() {
+        return Type.ATTRIBUTE;
+    }
+
+    /**
+     * Returns local part.
+     * @return local part
+     */
+    // -@cs[SimpleAccessorNameNotation] Overrides method from the base class.
```

Comment:
@romani 
Looks like the [feature request](https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/166#issuecomment-47854326) already opened,but is not implemented.

@timurt 
Please, add a link to the issue to the suppression comment.